caput/memh5.py

```diff
@@ -2569,6 +2576,9 @@ def _copy_from_file(h5group, memgroup, selections=None):
                         h5group, key, axis=distributed_axis, comm=comm, sel=selection
                     )
 
+                    # only needed until fixed: https://github.com/mpi4py/mpi4py/issues/177
+                    pdata = ensure_native_byteorder(pdata)
```

Comment:
I'm pretty sure this call is the one that's causing the test failures, but it actually isn't needed as `.from_hdf5` will guarantee that pdata has a explicit native endianness. Maybe true removing it, and see if things work in that case?